metadeploy/api/jobs.py

```diff
@@ -304,28 +305,23 @@ def create_scratch_org(*, plan_id, org_name, result_id):
         )
         async_to_sync(preflight_started)(org, preflight_result)
         preflight(preflight_result.pk)
-
-    # @@@ TODO: start installation job automatically if:
-    #   - Plan has no preflight
-    #   - Plan has no optional steps
-    # job = Job.objects.create(
-    #     user=None,
-    #     plan=plan,
-    #     organization_url=org_config.instance_url,
-    #     is_public=True,
-    #     org_id=scratch_org_config.config["org_id"],
-    # )
-
-    # rq_job = run_flows.delay(
-    #     plan=plan,
-    #     skip_steps=[],
-    #     organization_url=org_config.instance_url,
-    #     result_class=Job,
-    #     result_id=job.id,
-    # )
-    # job.job_id = rq_job.id
-    # job.enqueued_at = rq_job.enqueued_at
-    # job.save()
+    elif plan.required_step_ids.count() == plan.steps.count():
+        # Start installation job automatically if both:
+        # - Plan has no preflight
+        # - All plan steps are required
+        with transaction.atomic():
+            job = Job.objects.create(
+                user=None,
+                plan=plan,
+                organization_url=org_config.instance_url,
+                org_id=scratch_org_config.config["org_id"],
+                full_org_type=ORG_TYPES.Scratch,
+            )
+            job.steps.set(plan.steps.all())
+        # TODO: This is already called on `save()`, but the new Job isn't in the
+        # database yet because it's in an atomic transaction.
+        job.push_to_org_subscribers(is_new=True)
```

Comment:
@wlonk I'm curious if you have thoughts here. `job.push_to_org_subscribers` is already called automatically by the initial `job.save()` method, but my hand-testing revealed that in the eventual `notify_org_result_changed` fn, `Job.objects.all()` did not include the new triggering job -- I think likely because this all is `atomic` and has not been committed to the DB yet. I think this solution functionally works, but it does result in two subsequent WS events for the same job, and I'm not sure if there's a better way.

Reply:
It looks OK to me; without diving into debug it in detail, that analysis of the the problem behavior makes sense. Of course, I have pretty consistently been more OK with tradeoffs that result in spurious WS events than you have been, so that it looks OK to me is, well, maybe something to take with a grain of salt there. But OK for now, in any case!